test/python/scripts/test_st_submitjob.py

```diff
@@ -0,0 +1,269 @@
+    and "STREAMS_PASSWORD" in os.environ,
+    "requires Streams REST API setup",
+)
+class TestSubmitJob(unittest.TestCase):
+    def _submitjob(self, args, sab=None):
+        args.insert(0, "--disable-ssl-verify")
+        args.insert(1, "submitjob")
+        if sab:
+            args.insert(2, sab)
+        else:
+            topo = Topology()
+            topo.source([1])
+            cfg = {}
+            cfg[ConfigParams.SSL_VERIFY] = False
+            src = submit("BUNDLE", topo, cfg)
+            sab_path = src["build"]["artifacts"][0]["location"]
+            args.insert(2, sab_path)
+            self.files_to_remove.append(sab_path)
+        rc, val = streamtool.run_cmd(args=args)
+        return rc, val
+
+    def setUp(self):
+        self.instance = os.environ["STREAMS_INSTANCE_ID"]
+        self.username = os.environ["STREAMS_USERNAME"]
+        self.stringLength = 10
+        self.jobs_to_cancel = []
+        self.files_to_remove = []
+        self.name = "TEST__" + uuid.uuid4().hex.upper()[0 : self.stringLength]
+
+    def tearDown(self):
+        for job in self.jobs_to_cancel:
+            job.cancel(force=True)
+
+        self.files_to_remove.extend(glob.glob("./test_st_submitjob.*.json"))
+
+        for file in self.files_to_remove:
+            if os.path.exists(file):
+                os.remove(file)
+
+    # Check --jobname option
+    def test_submitjob_name(self):
+        rc, job = self._submitjob(args=["--jobname", self.name])
+
+        self.jobs_to_cancel.extend([job])
+        self.assertEqual(rc, 0)
+        self.assertEqual(job.name, self.name)
+
+    # Check --jobgroup option
+    def test_submitjob_group(self):
+
+        # JobGroup name doesn't exist, error should be printed to stderr
+        output, error, rc = self.get_output(
+            lambda: self._submitjob(args=["--jobgroup", str(self.name)])
+        )
+
+        error = error.splitlines()
+
+        if not any(
+            "500 Server Error: Internal Server Error for url" in s for s in error
+        ):
+            self.fail("jobgroup doesn't already exist, should throw 500 error")
+
+        # Check no jobGroup specified results in default jobGroup
+        rc, job = self._submitjob([])
+
+        self.jobs_to_cancel.extend([job])
+        jobgroup = job.jobGroup.split("/")[-1]
+
+        self.assertEqual(rc, 0)
+        self.assertEqual(jobgroup, "default")
+
+    # Check --jobConfig option
+    def test_submitjob_config(self):
+        jc = JobConfig(job_name=self.name)
+        my_file = "jobconfig.json"
+        with open(my_file, "w") as f:
+            json.dump(jc.as_overlays(), f)
+
+        rc, job = self._submitjob(args=["--jobConfig", my_file])
+
+        self.jobs_to_cancel.extend([job])
+        self.files_to_remove.append(my_file)
+
+        self.assertEqual(rc, 0)
+        self.assertEqual(job.name, self.name)
+
+    # Check --outfile option
+    def test_submitjob_outfile(self):
+        my_file = self.name + ".txt"
```

Comment:
Please use a temp file to avoid multiple runs of the test clashing with each other.